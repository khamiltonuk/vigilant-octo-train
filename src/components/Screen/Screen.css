.screen {
  background: #7ba89e;
  margin-bottom: 25px;
  position: relative;
}
.screen button {
  position: absolute;
  top: 0;
  right: 0;
  height: 100%;
}
input {
  background: none;
  min-height: 60px;
  text-align: right;
  border: 0;
  padding: 10px 90px 10px 10px;
  font-size: 2em;
  width: 100%;
}
