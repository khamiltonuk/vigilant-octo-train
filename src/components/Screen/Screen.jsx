import React from "react";
import "./Screen.css";

export default function Screen(props) {
  function handleChange(e) {
    props.keypadPress(e.target.value);
  }
  function handleSubmit(e) {
    console.log(1);
    e.preventDefault();
    if (props.displayValue !== "0" && props.displayValue !== "") {
      console.log(2);
      props.enterNumber(props.displayValue);
    }
  }
  const calculatedFontSize =
    props.displayValue && props.displayValue.lenght > 10 ? "1em" : "2em";
  return (
    <form className="screen" onSubmit={handleSubmit}>
      <input
        pattern="\d*"
        style={{ fontSize: calculatedFontSize }}
        onChange={e => handleChange(e)}
        onFocus={props.wipeInput}
        autoFocus="true"
        data-q="input-field"
        value={props.displayValue}
      />
      <button type="submit" data-q="input-number">
        Input number
      </button>
    </form>
  );
}
