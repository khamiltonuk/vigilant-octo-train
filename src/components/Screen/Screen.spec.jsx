import React from "react";
import { shallow, simulate } from "enzyme";
import Screen from "./Screen";
import toJson from "enzyme-to-json";

const props = {
  displayValue: jest.fn(),
  enterNumber: jest.fn(),
  keypadPress: jest.fn(),
  displayValue: "69"
};

describe("Screen", () => {
  test("should return a button", () => {
    const wrapper = shallow(<Screen {...props} />);
    const button = wrapper.find("button");
    expect(button.length).toBe(1);
  });

  test("should match the snapshot", () => {
    const wrapper = shallow(<Screen {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test.skip("should submit form when submit button is clicked", () => {
    const wrapper = shallow(<Screen {...props} />);
    const button = wrapper.find("[data-q='input-number']");
    button.simulate("click");
    expect(props.enterNumber).toHaveBeenCalledWith(5);
  });

  test.skip("should...u", () => {
    const wrapper = shallow(<Screen {...props} />);
    const input = wrapper.find("[data-q='input-field']");
    console.log(input.debug());
    input.simulate("change", { currentTarget: { value: 4 } });
    expect(props.enterNumber).toHaveBeenCalledWith(5);
  });
});
