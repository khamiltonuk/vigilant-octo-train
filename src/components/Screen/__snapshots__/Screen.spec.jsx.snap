// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen should match the snapshot 1`] = `
<form
  className="screen"
  onSubmit={[Function]}
>
  <input
    autoFocus="true"
    data-q="input-field"
    onChange={[Function]}
    pattern="\\\\d*"
    style={
      Object {
        "fontSize": "2em",
      }
    }
    value="69"
  />
  <button
    data-q="input-number"
    type="submit"
  >
    Input number
  </button>
</form>
`;
